File: LMIS/LMIS.Web/Modules/ServerTypes/InventoryManage/DownBookshelfForm.ts
import { StringEditor, LookupEditor, DateEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface DownBookshelfForm {
    BookId: StringEditor;
    BookshelfId: StringEditor;
    OperateUserId: LookupEditor;
    CreateTime: DateEditor;
    UpdateTime: DateEditor;
}

export class DownBookshelfForm extends PrefixedContext {
    static readonly formKey = 'InventoryManage.DownBookshelf';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!DownBookshelfForm.init)  {
            DownBookshelfForm.init = true;

            var w0 = StringEditor;
            var w1 = LookupEditor;
            var w2 = DateEditor;

            initFormType(DownBookshelfForm, [
                'BookId', w0,
                'BookshelfId', w0,
                'OperateUserId', w1,
                'CreateTime', w2,
                'UpdateTime', w2
            ]);
        }
    }
}

File: LMIS/LMIS.Web/Modules/InventoryManage/DownBookshelf/DownBookshelfForm.cs
using Serenity.ComponentModel;
using System;

namespace LMIS.InventoryManage.Forms;

[FormScript("InventoryManage.DownBookshelf")]
[BasedOnRow(typeof(DownBookshelfRow), CheckNames = true)]
public class DownBookshelfForm
{
    public long BookId { get; set; }
    public long BookshelfId { get; set; }
    public long OperateUserId { get; set; }
    public DateTime CreateTime { get; set; }
    public DateTime UpdateTime { get; set; }
}